lib/SpecializeImageTypes.cpp

```diff
@@ -143,8 +143,15 @@ Type *SpecializeImageTypesPass::RemapType(Argument *arg) {
 Type *SpecializeImageTypesPass::RemapUse(Value *value, unsigned operand_no) {
   if (CallInst *call = dyn_cast<CallInst>(value)) {
     auto called = call->getCalledFunction();
-    if (IsSampledImageRead(called) || IsUnsampledImageRead(called) ||
-        IsImageWrite(called)) {
+    auto func_info = Builtins::Lookup(called);
+    // TODO(sjw): should this handle half types also?
```

Comment:
read_imageh is converted in ReplaceOpenCLBuiltins with read_imagef and a conversion as Vulkan requires the sampled type to be a 32-bit scalar. Similar for write_imageh. So those shouldn't appear at this point in the flow.